src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -492,7 +455,9 @@ impl<'a, 'tcx> Visitor<'tcx> for Qualifier<'a, 'tcx, 'tcx> {
         match *place {
             Place::Local(ref local) => self.visit_local(local, context, location),
             Place::Static(ref global) => {
-                self.add(Qualif::STATIC);
+                if !(self.mode == Mode::Static || self.mode == Mode::StaticMut) {
```

Comment:
Can you add a comment above this `if` saying that only statics can refer to other statics?